### ISSUE TITLE
[content] Purpose-built railgun ships (not Hai railguns)

### ISSUE THREAD

**Issue by Taejang** [endless-sky/endless-sky#6383]
### Request Description

Railguns are standard sci fi fare. The Hai already use one in game, albeit a very small one on drones. I don't see why the Hai or someone else wouldn't have scaled this up for use on larger vessels. EVN had several calibers of railguns to fill the long-distance niche currently handled by the particle cannon in ES. My proposal here is different than either the EVN or Hai versions. For this request, I focus on it as a human concept, but there's no reason some other faction couldn't have it instead (or also).

### Related

I don't see a discussion on this already, at least not as I envision it.

### Request Details

The particle cannon currently performs the function EVN's railguns were designed for- long distance fighting. Rather than introduce a niche weapon with even longer range, or create something that competes directly with the particle cannon, I envision something more akin to the HALO universe's MAC weapons (or some Mass Effect ships, or really a lot of other science fiction vessels), where a ship is designed around the railgun. In this context, it would fire slowly as a fixed gun (never a turret), require more energy than most other human weapons, have long range and do a ton of damage- provided you can hit your target.

A light warship class could be designed to incorporate a single such railgun, while a medium or large warship could mount two or perhaps three. Each would have limited gun emplacements- the light warship may only have the single gun mount filled by the railgun, plus one or two turrets (default loadout would make these anti-missile turrets). The railgun can be visually incorporated into the design.

To fulfill this vision, the railgun would have to be restricted to railgun-capable ships. The ships themselves ideally should have a property allowing them to make use of the outfit so other ships cannot mount them.

It would be fun to see railgun ships be part of a campaign. For example, the ships could be developed during the Navy or pirate campaigns.

### Alternative Approaches

The ship+outfit could be designed such that a railgun is allowed to be installed on another ship, but is impractical. For example, the intended ship could have very low drag, while the railgun itself has high drag and requires a lot of outfit space- you _could_ put one on something else, but it would really slow you down and make it impossible to use your other gun mounts, making it strictly worse than using a ship designed around the railgun.

Twiddling with heat generation, mass, etc could further balance (or restrict) the idea of such a powerful gun being installed on another ship.

Likewise, you _could_ sell the railgun from a ship designed for it, but you'd have limited gun mounts, no fighter bays, little cargo space, etc, and thus the ship is strictly subpar compared to others if you remove the railgun. This may require making the railgun take a lot of weapon outfit space but little general outfit space- otherwise, you could sell the railgun and fill the ship with bunk rooms that thematically wouldn't make sense.

In short, the design could be such that a custom "railgun-capable" property is not needed, though it would require a lot more balancing.

**Comment by Chopliver19** [comment]
What if it was a small type of fighter that is essentially the railgun? P.S. I love the MAC gun from Halo!

**Comment by Zitchas** [comment]
To be fair, the railgun in EVN had *vastly* superior range to anything currently found in ES's human space. Having something extremely long range but very slow firing is a niche that isn't filled by anything in ES currently (other than the Archon's AMC)

**Comment by Taejang** [comment]
> What if it was a small type of fighter that is essentially the railgun? P.S. I love the MAC gun from Halo!
That's essentially what the Hai one is. I haven't fought against it much, no idea if it's any good or not.

> To be fair, the railgun in EVN had _vastly_ superior range to anything currently found in ES's human space. Having something extremely long range but very slow firing is a niche that isn't filled by anything in ES currently (other than the Archon's AMC)
Yeah, EVN railguns were awesome. I kind of assumed it was a style choice not to include them in ES.

**Comment by Zitchas** [comment]
Well, it is hard to say. I haven't seen anything suggesting it was deliberately omitted, but we (the community) did basically take over the project from the solo developer part way through. It could just be that he was focused on close quarters combat and hadn't done too much work on long range options yet. (case in point, the carrier acts in a decidedly un-carrier fashion)

**Comment by vitalchip** [comment]
My opinion is that it's a visual choice, partly informed by the support for low screen resolutions. Long range weapons would almost always be firing at targets off screen and likewise you'd not see the ship firing at you.

Star Trek phaser range is supposed to be 300,000 Km and yet we *see* this:
![main-qimg-97c516808c329072e03203dfdd80bd10-c](https://user-images.githubusercontent.com/65342577/144304417-ad7322db-8b37-4fd6-aaba-b29d7ab95806.jpeg)
..

**Comment by Zitchas** [comment]
Well, it might be worth taking a lesson from EVN on this: *most* weapons should be fairly short range. We've got a bit more flexibility thanks to zooming, but most should still be in the <600 range.

That being said, there is room for long range artillery. I won't say that EVN's railguns were spectacularly effective, but they were both environmental and fun. Perhaps a key detail is that while they felt like they could materially affect combat, they were sufficiently slow that it was really hard for them to actually *win* combat. As in, even a big ship loaded with nothing but railguns tended to fair very poorly because they just weren't fast enough (or high enough DPS) to outperform a ship even a category lower than them equipped with regular weapons.

In ES terms, a heavy warship loaded with EVN's railguns would lose to a medium warship armed with blasters. Even in an AI vs AI situation. 

However, a carrier with primarily railguns actually made a very good *support* ship, where the fighters and escorts did the fighting, and the carrier stayed at long range. Good tactics, and effective. And if a ship got pinned down so that railgun fire could actually hit... Quite useful.

**Comment by Taejang** [comment]
> The pirate campaign that @Galaucus and some others are working is planned to include a Massive railgun esque turret.

That is an awesome looking turret, and a fun idea.

**Comment by Galaucus** [comment]
Our plans are that railguns will have 50% shield piercing and deal fairly apocalyptic damage at a slow rate of fire. This is balanced by:
-Each will need a dedicated battery installation just to fire a full salvo
-You'll also probably want a reactor *for each* to keep up sustained barrages
-400mm ammo costs 10,000c/ea

The 400mm turrets will only really be usable by battlestations and as single installations on dedicated ship platforms.

We will, however, also be introducing 150mm turrets for use on standard ships. Same shield piercing, but much lower performance and operating costs.

**Comment by Zitchas** [comment]
Well, there's pro's and con's to fighters being hit by capital class weaponry. For example, when acting as screening. sacrificing one fighter to take out an AMC blast (or a heavy rocket, for that matter) that would have otherwise hit the carrier itself... That's a fair trade.

**Comment by Galaucus** [comment]
> Well, there's pro's and con's to fighters being hit by capital class weaponry. For example, when acting as screening. sacrificing one fighter to take out an AMC blast (or a heavy rocket, for that matter) that would have otherwise hit the carrier itself... That's a fair trade.

Maybe if it were something that could be toggled on the fighter itself? Ask it to screen or not to screen, and it would avoid or ram into shots accordingly. But... Having fighters tanking for a heavy warship seems like a very bad reversal of roles. I feel like ideally you'd want that sort of screening done by beefy frigates which can cycle in-out as shields are depleted and recharged.

**Comment by Zitchas** [comment]
Having the option to cycle strategies would be awesome. Especially if we could set it on a per fighter or per group basis.

**Comment by Taejang** [comment]
The way I see it, fighters and drones should ideally have the following apply to them:

1. All beams and projectiles ignore them as if they were allied units unless the firing ship is actually targeting the fighter/drone
2. Secondary weapons do not ignore them. You would almost never want a fighter screening its carrier from lasers, slicers, etc, but torpedoes? A nuke? Yes please.
3. Alternatively, it could be a weapon property applied to each weapon. "Does ignore fighters" and "Does not ignore fighters", with the understanding that if you actually target the fighter, _all_ weapons 'do not ignore fighters'

I don't personally like the idea of trying to individually order fighters, even groups of fighters, to screen or not screen. That kind of complexity doesn't mesh well with ES's hectic combat style, in my opinion. In most firefights, I can't even tell which fighters are which. Ordering them all to screen, or all not to? That could make sense.

**Comment by Galaucus** [comment]
Could have drones screen, but not fighters. If I were a fighter pilot and my commander told me "park here and eat torpedoes", I'd join the other team.

If we use an attribute to track what ship classes a particular shot can interact with, then we could have it that way.

**Comment by yafflehk** [comment]
> Could have drones screen, but not fighters. If I were a fighter pilot and my commander told me "park here and eat torpedoes", I'd join the other team.
> 
> If we use an attribute to track what ship classes a particular shot can interact with, then we could have it that way.

But the job of a fighter is to make it so unpleasant and dangerous to get inside of torpedo range that nobody does it. If you can shoot a carrier you've won the battle because the rest of the carrier group is sunk, disabled or in the wrong place. You don't have to eat the torpedo, just make sure nobody can fire one, or if they are firing them they are doing so very much in harms way. (Can we have the inaccuracy of weapons increase if the platform is under fire?)

**Comment by Galaucus** [comment]
Hm. Or tweak fighter-based weapons so that they can automatically do "antimissile" shots in a narrow frontal arc, just like a point defense turret would. That way fighter screens would be useful without relying on them physically body-blocking incoming fire.

**Comment by yafflehk** [comment]
Or make fighter shields bigger? Like three or four times the size of the ship. It won't make fighters any harder to kill because the hull target will stay the same size and the shield will have the same strength, but fighters will have a much greater "Screening" effect. (Unless I totally don't understand how shields are implemented in the game, which is pretty likely).

**Comment by Zitchas** [comment]
> Hm. Or tweak fighter-based weapons so that they can automatically do "antimissile" shots in a narrow frontal arc, just like a point defense turret would. That way fighter screens would be useful without relying on them physically body-blocking incoming fire.

This is where I wish that missiles were actual things that were treated like ships in that they could be targeted and affected by certain categories of weapons. Like EVN did it, actually. EVN's anti-missile would actually auto-fire on fighters too, because it was an air-defense weapon rather than a "only useful against missiles" weapon. 

It wasn't used in game, but from what I understand any weapon could be given the "anti-air" attribute, which would cause it to hit missiles and auto-fire at them when possible, but otherwise acted according to whatever sort of weapon it was. Gun, projectile turret, beam turret, whatever. In ES all anti-missiles are basically  beam turrets. In EVN most anti-missiles were things like "quad light blaster turrets" or "storm chainguns".

But anyway, yeah, giving fighters some sort of attribute that caused *all* primary weapons that they carry to hit all the projectile secondary weapons (missiles, torpedoes, anything anti-missile could hit) would give them interesting screening abilities.

**Comment by Taejang** [comment]
> This is where I wish that missiles were actual things that were treated like ships in that they could be targeted and affected by certain categories of weapons. Like EVN did it, actually. EVN's anti-missile would actually auto-fire on fighters too, because it was an air-defense weapon rather than a "only useful against missiles" weapon.
> 
> It wasn't used in game, but from what I understand any weapon could be given the "anti-air" attribute, which would cause it to hit missiles and auto-fire at them when possible, but otherwise acted according to whatever sort of weapon it was. Gun, projectile turret, beam turret, whatever. In ES all anti-missiles are basically beam turrets. In EVN most anti-missiles were things like "quad light blaster turrets" or "storm chainguns".

I really miss that. In ES, I fly around in my MaSSivE dEaTH SHiP o' DOoM and target big enemy ships, only to have all the fighters swarming around me like the pests they are... or I target stupid pests with every weapon I have. Sense this makes none.

The greater variety of defensive weapons in EVN was nice, too. In fact, I'm going to go make this a request right now.

**Comment by Zitchas** [comment]
Yeah, having the lightweight weapons (the afore-mentioned quad-light-blaster-turret and storm chaingun, among others) being both anti missiles and anti-fighter was great. It was rather like having turrets on ES's opportunistic & auto-fire mode, but *only* for those anti-tiny-target weapons.

This is sort of a related thing that I miss in ES. Good as the aformentioned opportunistic and autofire is, I do really wish I could limit it to the light & fast turrets that are dealing with the fighters, while I maintain control over the heavy turrets.

**Comment by movingpictures** [comment]
> To be fair, the railgun in EVN had _vastly_ superior range to anything currently found in ES's human space. Having something extremely long range but very slow firing is a niche that isn't filled by anything in ES currently (other than the Archon's AMC)

And the Emerald Sword's dragonflame.

**Comment by TomGoodIdea** [comment]
As other people noted, we already have a few such ships, so this issue can be closed. If you have more specific ideas, feel free to open a discussion or PR, or make a plugin implementing them.
